packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -183,11 +204,53 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
+      authorizationType: Lazy.string({
+        produce: () => {
+          if (this.iamEnabled) {
+            return 'AWS_IAM';
+          } else {
+            return authBindResult?.authorizationType ?? 'NONE'; // must be explicitly NONE (not undefined) for stack updates to work correctly
+          }
+        },
+      }),
       authorizationScopes,
     };
 
     const route = new CfnRoute(this, 'Resource', routeProps);
     this.routeId = route.ref;
   }
+
+  /**
+   * Enable IAM authorization on the route.
+   */
+  public enableIamAuthorization(): void {
+    if (this.authorizer) {
+      throw new Error('authorizationType has been set, so we cannot enable AWS_IAM authorization');
+    }
+
+    this.iamEnabled = true;
+  }
```

Comment:
I added `enableIamAuthorization()` because a user may want to enable IAM authorization on a route without granting invoke to any particular grantee. (As, for instance, when consuming this API from outside of the CDK.)

Reply:
How about creating an `HttpIamAuthorizer` instead?

The user will specify this against the `authorizer` attr if the IAM policy is defined outside.
If `grantInvoke` is called, you can set this automatically.

Reply:
@nija-at Ok. I've implemented. `HttpIamAuthorizer` in the `@aws-cdk/aws-apigatewayv2-authorizers` package. I didn't connect `grantInvoke` to `HttpIamAuthorizer` for fear of a dependency cycle between the two packages. It should be possible to move that route authorizer to the `@aws-cdk/aws-apigatwayv2` package so that we can re-use the feature. What do you think we should do in this case?

Reply:
For now, I've taken the liberty of moving the `HttpIamAuthorizer` out of the `-authorizers` package so `grantInvoke` can use the same code as the IAM Authorizer. We can revert this, pending the outcome of this discussion.

Reply:
I think this is the last point we need to work out. I'd rather have all authorizers in the `-authorizers` module. What's your idea here? How would you avoid the circular dependency?